cmake/CMakeLists.txt

```diff
@@ -353,6 +353,7 @@ file (STRINGS "${REPO_ROOT}/VERSION_NUMBER" ORT_VERSION)
 
 if (onnxruntime_ENABLE_TRAINING)
   set(onnxruntime_ENABLE_ATEN ON)
+  set(onnxruntime_ENABLE_TRAINING_OPS ON)
```

Comment:
I searched "--enable_training_ops" in our yaml files, then I found the only case this option is ON but onnxruntime_ENABLE_TRAINING=OFF is in ORT inference minimal build. I think probably it would be better to add a daily pipeline to covert to the non-minimal build cases.

Reply:
Also, you may try:

```
cmake_dependent_option(USE_FOO "Use Foo" OFF "NOT USE_ZOT" ON)
```
It means:
If USE_ZOT is false, this provides an option called USE_FOO that defaults to OFF. Otherwise, it sets USE_FOO to ON and hides the option from the user.

Reply:
I am not really sure why ENABLE_TRAINING_OPS was added in first place but I am planning to make sure of it to simplify creation of on device training builds. The idea is both full training builds and on device training builds can set ENABLE_TRAINING_OPS this way we can avoid ENABLE_TRAINING || ENABLE_TRAINING_ON_DEVICE in a lot of places. 

Right now getting the current pipelines to build should be sufficient. Once this PR is done I will create a another one to create a dedicated build for on device training and that will test the case where ENABLE_TRAINING_OPS is true but ENABLE_TRAINING is false.

Reply:
ENABLE_TRAINING_OPS  from its name is to control whether some kernels ops are built, I guess this is reason we have it separately. 

ENABLE_TRAINING_ON_DEVICE ideally should be used independent of  ENABLE_TRAINING or ENABLE_TRAINING_OPS, because there are limited of things that should be included for on device training. Currently ENABLE_TRAINING_ON_DEVICE  is duplicated with ENABLE_TRAINING a lot doing the check, once we do clean up, I assume this will be expected use ENABLE_TRAINING_ON_DEVICE and ENABLE_TRAINING  to control different things, right?

Reply:
ENABLE_TRAINING enables building code (such as the older `TrainingSession`, `OptimizerGraphBuidler` and many more) that might not be relevant in many cases (such as in the minimal builds).

ENABLE_TRAINING_OPS should control kernel building. So, non-training code (code that does not rely on TrainingSession, Training apis) can invoke kernels defined in the `training_op_defs` that may be relevant for them.

Reply:
Disregard ENABLE_TRAINING_ON_DEVICE macro in the context of this PR. This PR is only meant to clean the usage of ENABLE_TRAINING_OPS. 

ENABLE_TRAINING_OPS should control whether training ops are included in the build. ENABLE_TRAINING will include all the rest of the files specific to training. ENABLE_TRAINING_OPS should be set to true when ENABLE_TRAINING is set. This is the behavior I am aiming to achieve.